src/flakeref.rs

```diff
@@ -730,20 +691,18 @@ mod tests {
             }))
             .clone();
 
-        let parsed: FlakeRef = uri.try_into().unwrap();
-        let (rest, nommed) = FlakeRef::parse(uri).unwrap();
+        let nommed = FlakeRef::parse(&mut uri).unwrap();
 
-        assert_eq!("", rest);
-        assert_eq!(expected, parsed);
+        assert_eq!("", uri);
         assert_eq!(expected, nommed);
     }
 
     #[test]
     #[should_panic(
-        expected = "called `Result::unwrap()` on an `Err` value: UnknownUriType(\"gt+https\")"
+        expected = "called `Result::unwrap()` on an `Err` value: Backtrack(ContextError { context: [Label(\"FlakeRefType\")], cause: None })"
```

Comment:
followup issue: "gt+" could be a cut-error if +, :// or empty is seen following a failed resource-type issue